wandb/sdk/launch/builder/kaniko_builder.py

```diff
@@ -328,6 +348,31 @@ def _create_kaniko_job(
                     value=self.registry.environment.region,
                 )
             ]
+        if isinstance(self.environment, AzureEnvironment):
+            # Use the core api to check if the secret exists
+            try:
+                core_client.read_namespaced_secret(
+                    "azure-storage-access-key",
+                    "wandb",
+                )
+            except kubernetes.kubernetes.client.ApiException as e:
```

Comment:
Are there other possible exceptions that could occur here?

Reply:
Mmm yeah I'm just going to blanket this.